polaris-service/src/main/java/io/polaris/service/catalog/BasePolarisCatalog.java

```diff
@@ -287,6 +287,20 @@ private Set<String> getLocationsAllowedToBeAccessed(TableMetadata tableMetadata)
     Set<String> locations = new HashSet<>();
     locations.add(concatFilePrefixes(basicLocation, "data/", "/"));
     locations.add(concatFilePrefixes(basicLocation, "metadata/", "/"));
+    if (tableMetadata
```

Comment:
This is a good but I wonder if we need a way to add even more locations, the basic issue is this only represents the "latest" location that data or metadata could be written to and a user could have set a different value to these properties in the past.

Reply:
This is the purpose of the `allowedLocations` field of the `storageConfiguration`. Currently, we can't set table-level StorageConfiguration, but part of the reason for implementing the `StorageConfigurationOverride` is to help pave the way toward that.